lib/std/unicode.zig

```diff
@@ -196,22 +196,109 @@ pub fn utf8CountCodepoints(s: []const u8) !usize {
     return len;
 }
 
-pub fn utf8ValidateSlice(s: []const u8) bool {
+/// Returns true if the input consists entirely of UTF-8 codepoints
+pub fn utf8ValidateSlice(input: []const u8) bool {
+    var remaining = input;
+
+    // Fast path. Check for and skip 8 bytes of ASCII characters per iteration.
+    while (remaining.len >= 8) {
+        const first32 = mem.readIntLittle(u32, remaining[0..4]);
+        const second32 = mem.readIntLittle(u32, remaining[4..8]);
+        if ((first32 | second32) & 0x80808080 != 0) {
+            // Found a non ASCII byte
+            break;
+        }
+        remaining = remaining[8..];
+    }
```

Comment:
How about using `usize`?

```suggestion
    while (remaining.len >= @sizeOf(usize)) {
        const first = mem.readIntLittle(usize, remaining[0..@sizeOf(usize)]);
        const mask: usize = @bitCast([1]u8{0x80} ** @sizeOf(usize));
        if (first & mask != 0) {
            // Found a non ASCII byte
            break;
        }
        remaining = remaining[@sizeOf(usize)..];
    }
```

Or maybe vectors?

```suggestion
    const V = @Vector(std.simd.suggestVectorSize(usize) orelse 1, usize);
    while (remaining.len >= @sizeOf(usize) * V.len) {
        const first: V = remaining[0..V.len].*;
        const swapped = switch (builtin.cpu.arch.endian()) {
            .Big => @byteSwap(first),
            .Little => first,
        };
        const reduced = @reduce(.Or, swapped);
        const mask: usize = @bitCast([1]u8{0x80} ** @sizeOf(usize));
        if (reduced & mask != 0) {
            // Found a non ASCII byte
            break;
        }
        remaining = remaining[@sizeOf(usize) * V.len..];
    }
```

Reply:
I think:
```zig
 const first: V = remaining[0..V.len].*;
 ```
 
 should be:
 
 ```zig
 const first: V = @bitCast(remaining[0..(@sizeOf(usize) * V.len)].*);
```

I added a new test case to cover this, iterate through a 0->550 length invalid string, to make sure the fast path is properly chunking the input.

Reply:
I see the mistake I made. Instead of changing it to bitcast, I think you should change the vector element to be a `u8`.

Reply:
Well, I wrote that, but the usize elements with the bitcast seems to perform better at least on my machine, so if that works, then cool :+1:

Reply:
This has the benefit that if `V_len` is 1, it still processes `usize` at a time.